src/main/scala/org/ergoplatform/nodeView/history/ErgoHistoryReader.scala

```diff
@@ -105,7 +105,13 @@ trait ErgoHistoryReader
     */
   private def commonPoint(headers: Seq[Header]): Option[Header] = {
     headers.find { h =>
-      contains(h.id)
+      contains(h.id) &&
+        historyStorage.modifierById(h.id).exists {
+          case Header(_, _, _, _, _, _, _, _, _, pow, _, _) =>
+            h.powSolution == pow
```

Comment:
No need for such check (pow already reflected in id, which collision-resistance is cryptographically strong).